lib/shopify_app/login_protection.rb

```diff
@@ -42,5 +42,11 @@ def close_session
       redirect_to login_path
     end
 
+    def login_path(params = {})
+      main_app.login_path(params)
+    rescue NoMethodError => e
+      shopify_app.login_path(params)
+    end
```

Comment:
for some reason using `.try` here didn't work. It would work the first time but if you tried to login in to a different shop like described in #134 it would raise `NoMethodError` again. I opted for this fix which always works.